File: Data Structures & Algorithms/jump-game/submission-0.java
class Solution {
    public boolean canJump(int[] arr) {
        int n = arr.length;

        boolean[] dp = new boolean[n];
        dp[0] = true;

        for (int i = 1 ; i < n; i++) {

            for (int j = 0; j < i; j++) {
                if (dp[j] && j + arr[j] >= i) {
                    dp[i] = true;
                    break;
                }
            }
        }

        return dp[n-1];
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, arr: List[int]) -> int:
        n = len(arr)
        minlen = n
        currlen = 1

        for i in range(n):
            if arr[i] != 1:
                minlen = min(minlen , currlen)
                currlen = 0
            else:
                currlen += 1

        return minlen